### PR TITLE
ManagedIdentityTokenResponse's expires_in can now be accepted in ISO 8601 date format

### LINKED ISSUE #7393: [Bug] Fix Date-Time Parsing in Token Expiration Calculation for Multiple Formats
### Core Library

MSAL.js (@azure/msal-node)

### Core Library Version

latest

### Public or Confidential Client?

Confidential

### Description

The method responsible for calculating token expiration (expires_on) in MSAL (Managed Identity) is not handling various date-time formats correctly. This leads to errors or incorrect results when processing expiration times in different formats provided by the Managed Identity service.

more info : https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4963 

### Error Message

_No response_

### MSAL Logs

_No response_

### Network Trace (Preferrably Fiddler)

- [ ] Sent
- [ ] Pending

### MSAL Configuration

```javascript
Added parsing logic for Unix timestamps, ISO 8601
```

### Relevant Code Snippets

```javascript
Added parsing logic for Unix timestamps, ISO 8601
```

### Reproduction Steps

Added parsing logic for Unix timestamps, ISO 8601

### Expected Behavior

Added parsing logic for Unix timestamps, ISO 8601

### Identity Provider

Entra ID (formerly Azure AD) / MSA

### Browsers Affected (Select all that apply)

None (Server)

### Regression

_No response_

### PR DESCRIPTION
**Pull request by Robbie-Microsoft**
Fixes #7393

### CODE REVIEW

File: lib/msal-common/src/utils/TimeUtils.ts [hunk 1]
```diff
@@ -48,3 +48,15 @@ export function wasClockTurnedBack(cachedAt: string): boolean {
...
 }
+
+/**
+ * @internal
+ * Checks if a given date string is in ISO 8601 format.
+ *
+ * @param dateString - The date string to be checked.
+ * @returns boolean - Returns true if the date string is in ISO 8601 format, otherwise false.
+ */
+export function isIso8601(dateString: string): boolean {
```
Comment (bgavrilMS) [inline lib/msal-common/src/utils/TimeUtils.ts]: move to msal-node for simplicity?

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts [hunk 2]
```diff
@@ -84,6 +85,15 @@ export abstract class BaseManagedIdentitySource {
     ): ServerAuthorizationTokenResponse {
         let refreshIn, expiresIn: number | undefined;
         if (response.body.expires_on) {
+            // if the expires_on field in the response body is a string and in ISO 8601 format, convert it to a Unix timestamp (seconds since epoch)
+            if (
```
Comment (bgavrilMS) [inline lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: nit: strange use of paranthesis. why not 

```
if (condition && condition) {
    do_stuff();
}
```

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts [hunk 3]
```diff
@@ -84,6 +85,15 @@ export abstract class BaseManagedIdentitySource {
...
+                typeof response.body.expires_on === "string" &&
+                isIso8601(response.body.expires_on)
```
Comment (bgavrilMS) [inline lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: Avoid double parsing. 

For example in C# the pattern used is `bool TryParseIso8601Date(string input, out Date parsedDate)`. I guess in NodeJS the pattern is more like `ParseIso8601(date: string) : Date | null` or smth like that (but not undefined)

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
Avoid double parsing.

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 2 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: ? There's other pre-existing logic already happening in this if statement that depends on `response.body.expires_on` existing.

// compute refresh_in as 1/2 of expires_in, but only if expires_in > 2h

File: lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts [hunk 4]
```diff
@@ -681,6 +682,27 @@ describe("Acquires a token successfully via an IMDS Managed Identity", () => {
             ).toBe(false);
         }, 10000); // double the timeout value for this test because it waits two seconds in between the acquireToken call and the cache lookup
 
+        test("ensures an ISO 8601 date returned by the Managed Identity is converted to a Unix timestamp (seconds since epoch)", async () => {
```
Comment (gladjohn) [inline lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts:685]: can you extend the tests to cover few more cases, see MSAL NET test coverage [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dec281197a8f49f343263247dbff3c3090185736/tests/Microsoft.Identity.Test.Unit/UtilTests/DateTimeHelperTests.cs)

**MSAL .NET tests validate:**

- **Unix timestamps** (seconds since epoch) for past, present, and future values.
- **ISO 8601 format** (e.g., 2024-10-18T19:51:37.0000000+00:00).
- **Common date formats** (e.g., MM/dd/yyyy HH:mm:ss, yyyy-MM-dd HH:mm:ss).
- **Invalid formats** (which should throw an exception).

**also consider testing:**

- **Time zone variations** Example: 2024-10-18T19:51:37-07:00 (time zones with offsets).
- **Negative timestamps** Example: -1697490590 (handling of past timestamps).
- **Extreme future timestamps** Example: 31556995200 (1000 years into the future).
- **Fractional Unix timestamps** Example: 1697490590.123 (valid timestamps with milliseconds).

Comment (Robbie-Microsoft) [reply lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts:685]: Unix timestamps and their test coverage already exist. The purpose of this PR is to add functionality for ISO 8601 timestamps.

This PR simply converts an ISO 8601 date to a unix timestamp.

We can open a new PR if you'd like to see additional test coverage for unix timestamp functionality.

Comment (gladjohn) [reply lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts:685]: Can you add tests for ISO 8601 edge cases like time zones (+05:30, -07:00), fractional seconds (.123Z), date-only formats (2024-10-18), and invalid values (25:00:00)?

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 3 above)
Comment (bgavrilMS) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: Not resolved. Double parsing happens at line 94. It's not a blocking comment.

**Review by bgavrilMS — APPROVED** [review]

File: lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts (hunk 3 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts]: I still don't understand what you're asking. I reached out to you on Teams.

File: lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts (hunk 4 above)
Comment (gladjohn) [reply lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts:685]: here is the data @Robbie-Microsoft 

```js
// Example test data for ISO 8601 variants:
const testDates = [
  {
    input: "2025-01-14T19:51:37-07:00",
    description: "Valid ISO 8601 with timezone offset"
  },
  {
    input: "2025-01-14T19:51:37.123Z",
    description: "Valid ISO 8601 in UTC with fractional seconds"
  },
  {
    input: "2025-01-14T25:00:00Z",
    description: "Invalid time (25 is not a valid hour)"
  },
  {
    input: "2025-01-14T19:51:37Z",
    description: "Valid ISO 8601 in UTC with no fractional seconds"
  },
  {
    input: "2025-01-14",
    description: "Date-only (no time)"
  },
];
```

Comment (bgavrilMS) [reply lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts:685]: Comment not resolved?
